### PR TITLE
Add rubocop config

### PR DESCRIPTION
**Pull request by bastelfreak**
previously this gem only contained a rubocop config that's used in our modules. We didn't run rubocop for the ruby code in this exact gem. This is changed with this commit.

### CODE REVIEW

**Comment by ekohl** [comment]
Sounds good to me, but RuboCop is not happy with you

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# this alligns with our settings in voxpupuli-rubocop
+# but we cannot depend on that because it pulls in newer rubocop versions,
+# which would break our modules on legacy puppetserver
+AllCops:
+  NewCops: enable
+  DisplayCopNames: true
+  ExtraDetails: true
+  DisplayStyleGuide: true
+  TargetRubyVersion: '2.7'
```
Comment (ekohl) [inline .rubocop.yml:14]: RuboCop should be able to derive this from the gemspec, which I'd prefer.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+# Offense count: 1
+RSpec/MultipleExpectations:
+  Max: 4
+
+# Offense count: 3
+# This cop supports unsafe autocorrection (--autocorrect-all).
+# Configuration parameters: EnforcedStyle.
+# SupportedStyles: always, always_true, never
+Style/FrozenStringLiteralComment:
```
Comment (ekohl) [inline .rubocop_todo.yml:78]: I see you added it to `Rakefile`, but should we add these to other files as well?

**Review by ekohl — COMMENTED** [review]
I opened https://github.com/voxpupuli/voxpupuli-test/pull/137 to fix 2 TODOs.

File: .rubocop_todo.yml (hunk 2 above)
Comment (bastelfreak) [reply .rubocop_todo.yml]: done

File: .rubocop.yml (hunk 1 above)
Comment (bastelfreak) [reply .rubocop.yml:14]: that would be nice, yes

**Comment by bastelfreak** [comment]
@ekohl I merged your PR and rebased.

File: .rubocop.yml (hunk 1 above)
Comment (ekohl) [reply .rubocop.yml:14]:
```suggestion
```

File: lib/voxpupuli/test/rake.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
+# frozen_string_literal: true
+
 require 'puppetlabs_spec_helper/rake_tasks'
 
-PuppetLint.configuration.log_format = '%{path}:%{line}:%{check}:%{KIND}:%{message}'
+PuppetLint.configuration.log_format = '%<path>s:%<line>s:%<check>s:%<KIND>s:%<message>s'
```
Comment (ekohl) [inline lib/voxpupuli/test/rake.rb:5]: I honestly don't now if puppet-lint understands this format.

File: .rubocop.yml (hunk 1 above)
Comment (bastelfreak) [reply .rubocop.yml]: oh I got you wrong. You mean rubocop actually supports that, nice.

File: lib/voxpupuli/test/spec_helper.rb [hunk 4]
```diff
@@ -1,10 +1,10 @@
+# frozen_string_literal: true
+
 RSpec.configure do |config|
   # puppetlabs_spec_helper defaults to mocha but emits a deprecation warning
   # Vox Pupuli prefers rspec to avoid the deprecation warning unless explicitly
   # set
-  if config.instance_variable_get(:@mock_framework).nil?
-    config.mock_with :rspec
-  end
+  config.mock_with :rspec if config.instance_variable_get(:@mock_framework).nil?
```
Comment (ekohl) [inline lib/voxpupuli/test/spec_helper.rb]: This is now redundant so let's remove it instead: https://github.com/voxpupuli/voxpupuli-test/pull/138

File: lib/voxpupuli/test/rake.rb (hunk 3 above)
Comment (bastelfreak) [reply lib/voxpupuli/test/rake.rb:5]: It seems to work fine:
```
bastelfreak@bastelfreak-nb ~/code/modulesync_config/modules/voxpupuli/puppet-example $ git diff Gemfile manifests/init.pp
diff --git a/Gemfile b/Gemfile
index a4a3b20..3a8a46c 100644
--- a/Gemfile
+++ b/Gemfile
@@ -4,7 +4,8 @@
 source ENV['GEM_SOURCE'] || 'https://rubygems.org'

 group :test do
-  gem 'voxpupuli-test', '~> 7.0',   :require => false
+  #gem 'voxpupuli-test', '~> 7.0',   :require => false
+  gem 'voxpupuli-test', git: 'https://github.com/bastelfreak/voxpupuli-test', branch: 'rubocop2'
   gem 'coveralls',                  :require => false
   gem 'simplecov-console',          :require => false
   gem 'puppet_metadata', '~> 3.5',  :require => false
diff --git a/manifests/init.pp b/manifests/init.pp
index fbf7850..eacd9af 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -1,8 +1,6 @@
 # @summary An example class
 # @param file
 #   The file to manage
-# @param content
-#   The content in the file
 class example (
   Stdlib::Absolutepath $file = '/tmp/puppet-example',
   String[1] $content = 'Hello World!',
bastelfreak@bastelfreak-nb ~/code/modulesync_config/modules/voxpupuli/puppet-example $
```

**Review by rwaffen — APPROVED** [review]

File: lib/voxpupuli/test/rake.rb (hunk 3 above)
Comment (rwaffen) [reply lib/voxpupuli/test/rake.rb:5]: was asking myself the same (my rubocop plugin wanted to auto fix this every time i opend the file XD) . but if it is tested...

**Pull request merged**